src/main/java/com/minecolonies/coremod/blocks/BlockMinecoloniesRack.java

```diff
@@ -0,0 +1,449 @@
+        {
+            final IItemHandler handler = ((TileEntityRack) tileentity).getInventory();
+            InventoryUtils.dropItemHandler(handler, worldIn, pos.getX(), pos.getY(), pos.getZ());
+        }
+
+        super.breakBlock(worldIn, pos, state);
+    }
+
+    @Override
+    public boolean onBlockActivated(
+            final World worldIn,
+            final BlockPos pos,
+            final IBlockState state,
+            final EntityPlayer playerIn,
+            final EnumHand hand,
+            @Nullable final ItemStack heldItem,
+            final EnumFacing side,
+            final float hitX,
+            final float hitY,
+            final float hitZ)
+    {
+        /*
+        If the world is client, open the gui of the building
+         */
+        if (!worldIn.isRemote)
+        {
+            final Colony colony = ColonyManager.getColony(worldIn, pos);
+            final TileEntity tileEntity = worldIn.getTileEntity(pos);
+
+            if ((colony == null || colony.getPermissions().hasPermission(playerIn, Action.ACCESS_HUTS))
+                    && tileEntity instanceof TileEntityRack)
+            {
+                playerIn.openGui(MineColonies.instance, 0, worldIn, pos.getX(), pos.getY(), pos.getZ());
+                return true;
+            }
+        }
+        return false;
+    }
+
+    /**
+     * Called when a user uses the creative pick block button on this block
+     * @param state the state.
+     * @param target the target.
+     * @param world the world.
+     * @param pos the position.
+     * @param player the player.
+     * @return
+     */
+    @Override
+    public ItemStack getPickBlock(final IBlockState state, final RayTraceResult target, final World world, final BlockPos pos, final EntityPlayer player)
+    {
+        return new ItemStack(this, 1);
+    }
+
+    @NotNull
+    @Override
+    protected BlockStateContainer createBlockState()
+    {
+        return new BlockStateContainer(this, FACING, VARIANT);
+    }
+
+    @Override
+    public boolean hasTileEntity(final IBlockState state)
+    {
+        return true;
+    }
+
+    @Override
+    public TileEntity createTileEntity(final World world, final IBlockState state)
+    {
+        return new TileEntityRack();
+    }
+
+    /**
+     * @deprecated (Remove this as soon as minecraft offers anything better).
+     */
+    @Override
+    @Deprecated
+    public boolean isFullBlock(final IBlockState state)
+    {
+        return false;
+    }
+
+    /**
+     * Convert the given metadata into a BlockState for this Block.
+     *
+     * @deprecated (Remove this as soon as minecraft offers anything better).
+     */
+
+    /**
+     * Convert the BlockState into the correct metadata value.
+     *
+     * @deprecated (Remove this as soon as minecraft offers anything better).
+     */
+    @NotNull
+    @Override
+    @Deprecated
+    public IBlockState withRotation(@NotNull final IBlockState state, final Rotation rot)
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Do not forget to remove this deprecated code someday. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1133)